TestConfigAll is unstable

## Flaky Test

### Which jobs are failing
```
--- FAIL: TestConfigTestSuite/TestConfigAll (5.31s)
        config_test.go:180: 
            	Error Trace:	/home/runner/work/pd/pd/tests/server/config/config_test.go:180
            	            				/home/runner/work/pd/pd/tests/testutil.go:309
            	            				/home/runner/work/pd/pd/tests/testutil.go:270
            	            				/home/runner/work/pd/pd/tests/server/config/config_test.go:104
            	Error:      	Not equal: 
            	            	expected: &config.Config{ClientUrls:"http://127.0.0.1:35273", PeerUrls:"http://127.0.0.1:39959", AdvertiseClientUrls:"http://127.0.0.1:35273", AdvertisePeerUrls:"http://127.0.0.1:39959", Name:"pd1", DataDir:"/tmp/pd-tests1161716090", ForceNewCluster:false, EnableGRPCGateway:true, InitialCluster:"pd1=http://127.0.0.1:39959", InitialClusterState:"new", InitialClusterToken:"pd-cluster", Join:"", LeaderLease:3, Log:log.Config{Level:"warn", Format:"text", DisableTimestamp:false, File:log.FileLogConfig{Filename:"", MaxSize:0, MaxDays:0, MaxBackups:0}, Development:false, DisableCaller:false, DisableStacktrace:false, DisableErrorVerbose:true, Sampling:(*zap.SamplingConfig)(nil), ErrorOutputPath:""}, LogFileDeprecated:"", LogLevelDeprecated:"", MaxConcurrentTSOProxyStreamings:5000, TSOProxyRecvFromClientTimeout:typeutil.Duration{Duration:3600000000000}, TSOSaveInterval:typeutil.Duration{Duration:3000000000}, TSOUpdatePhysicalInterval:typeutil.Duration{Duration:50000000}, EnableLocalTSO:false, Metric:metricutil.MetricConfig{PushJob:"pd1", PushAddress:"", PushInterval:typeutil.Duration{Duration:15000000000}}, Schedule:config.ScheduleConfig{MaxSnapshotCount:0x40, MaxPendingPeerCount:0x40, MaxMergeRegionSize:0x14, MaxMergeRegionKeys:0x30d40, SplitMergeInterval:typeutil.Duration{Duration:3600000000000}, SwitchWitnessInterval:typeutil.Duration{Duration:3600000000000}, EnableOneWayMerge:false, EnableCrossTableMerge:true, PatrolRegionInterval:typeutil.Duration{Duration:10000000}, MaxStoreDownTime:typeutil.Duration{Duration:1800000000000}, MaxStorePreparingTime:typeutil.Duration{Duration:172800000000000}, LeaderScheduleLimit:0x4, LeaderSchedulePolicy:"count", RegionScheduleLimit:0xa, WitnessScheduleLimit:0x4, ReplicaScheduleLimit:0x40, MergeScheduleLimit:0x8, HotRegionScheduleLimit:0x4, HotRegionCacheHitsThreshold:0x3, StoreBalanceRate:0, StoreLimit:map[uint64]config.StoreLimitConfig{}, TolerantSizeRatio:2.5, LowSpaceRatio:0.8, HighSpaceRatio:0.7, RegionScoreFormulaVersion:"v2", SchedulerMaxWaitingOperator:0x5, DisableLearner:false, DisableRemoveDownReplica:false, DisableReplaceOfflineReplica:false, DisableMakeUpReplica:false, DisableRemoveExtraReplica:false, DisableLocationReplacement:false, EnableRemoveDownReplica:true, EnableReplaceOfflineReplica:true, EnableMakeUpReplica:true, EnableRemoveExtraReplica:true, EnableLocationReplacement:true, EnableDebugMetrics:false, EnableJointConsensus:true, EnableTiKVSplitRegion:false, Schedulers:config.SchedulerConfigs{config.SchedulerConfig{Type:"balance-region", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"balance-leader", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"balance-witness", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"hot-region", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"transfer-witness-leader", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"evict-slow-store", Args:[]string(nil), Disable:false, ArgsPayload:""}}, SchedulersPayload:map[string]interface {}{"balance-hot-region-scheduler":map[string]interface {}{"byte-rate-rank-step-ratio":0.05, "count-rank-step-ratio":0.01, "dst-tolerance-ratio":1.05, "enable-for-tiflash":"true", "forbid-rw-type":"none", "great-dec-ratio":0.95, "key-rate-rank-step-ratio":0.05, "max-peer-number":1000, "max-zombie-rounds":3, "min-hot-byte-rate":100, "min-hot-key-rate":10, "min-hot-query-rate":10, "minor-dec-ratio":0.99, "query-rate-rank-step-ratio":0.05, "rank-formula-version":"v2", "read-priorities":[]interface {}{"query", "byte"}, "split-thresholds":0.2, "src-tolerance-ratio":1.05, "strict-picking-store":"true", "write-leader-priorities":[]interface {}{"query", "byte"}, "write-peer-priorities":[]interface {}{"byte", "key"}}, "balance-leader-scheduler":map[string]interface {}{"batch":4, "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "balance-region-scheduler":map[string]interface {}{"name":"balance-region-scheduler", "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "balance-witness-scheduler":map[string]interface {}{"batch":4, "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "evict-slow-store-scheduler":map[string]interface {}{"evict-stores":[]interface {}{}, "recovery-duration":600}, "transfer-witness-leader-scheduler":interface {}(nil)}, HotRegionsWriteInterval:typeutil.Duration{Duration:600000000000}, HotRegionsReservedDays:0x7, MaxMovableHotPeerSize:512, EnableDiagnostic:true, EnableWitness:false, SlowStoreEvictingAffectedStoreRatioThreshold:0.3, StoreLimitVersion:"v1", HaltScheduling:false}, Replication:config.ReplicationConfig{MaxReplicas:0x5, LocationLabels:typeutil.StringSlice{"idc", "host"}, StrictlyMatchLabel:false, EnablePlacementRules:true, EnablePlacementRulesCache:false, IsolationLevel:""}, PDServerCfg:config.PDServerConfig{UseRegionStorage:true, MaxResetTSGap:typeutil.Duration{Duration:86400000000000}, KeyType:"table", RuntimeServices:typeutil.StringSlice{}, MetricStorage:"http://127.0.0.1:1234", DashboardAddress:"http://127.0.0.1:35273", TraceRegionFlow:true, FlowRoundByDigit:3, MinResolvedTSPersistenceInterval:typeutil.Duration{Duration:1000000000}, ServerMemoryLimit:0, ServerMemoryLimitGCTrigger:0.7, EnableGOGCTuner:false, GCTunerThreshold:0.6, BlockSafePointV1:false}, ClusterVersion:semver.Version{Major:4, Minor:0, Patch:0, PreRelease:"beta", Metadata:""}, Labels:map[string]string{}, QuotaBackendBytes:0x200000000, AutoCompactionMode:"periodic", AutoCompactionRetention:"1h", TickInterval:typeutil.Duration{Duration:500000000}, ElectionInterval:typeutil.Duration{Duration:3000000000}, PreVote:true, MaxRequestBytes:0x9600000, Security:configutil.SecurityConfig{TLSConfig:grpcutil.TLSConfig{CAPath:"", CertPath:"", KeyPath:"", CertAllowedCN:[]string(nil), SSLCABytes:[]uint8(nil), SSLCertBytes:[]uint8(nil), SSLKEYBytes:[]uint8(nil)}, RedactInfoLog:false, Encryption:encryption.Config{DataEncryptionMethod:"plaintext", DataKeyRotationPeriod:typeutil.Duration{Duration:604800000000000}, MasterKey:encryption.MasterKeyConfig{Type:"plaintext", MasterKeyKMSConfig:encryption.MasterKeyKMSConfig{KmsKeyID:"", KmsRegion:"", KmsEndpoint:""}, MasterKeyFileConfig:encryption.MasterKeyFileConfig{FilePath:""}}}}, LabelProperty:config.LabelPropertyConfig{}, WarningMsgs:[]string(nil), DisableStrictReconfigCheck:false, HeartbeatStreamBindInterval:typeutil.Duration{Duration:60000000000}, LeaderPriorityCheckInterval:typeutil.Duration{Duration:60000000000}, Logger:(*zap.Logger)(nil), LogProps:(*log.ZapProperties)(nil), Dashboard:config.DashboardConfig{TiDBCAPath:"", TiDBCertPath:"", TiDBKeyPath:"", PublicPathPrefix:"", InternalProxy:false, EnableTelemetry:false, EnableExperimental:false}, ReplicationMode:config.ReplicationModeConfig{ReplicationMode:"dr-auto-sync", DRAutoSync:config.DRAutoSyncReplicationConfig{LabelKey:"foobar", Primary:"", DR:"", PrimaryReplicas:0, DRReplicas:0, WaitStoreTimeout:typeutil.Duration{Duration:60000000000}, WaitRecoverTimeout:typeutil.Duration{Duration:0}, PauseRegionSplit:false}}, Keyspace:config.KeyspaceConfig{PreAlloc:[]string(nil), WaitRegionSplit:true, WaitRegionSplitTimeout:typeutil.Duration{Duration:30000000000}, CheckRegionSplitInterval:typeutil.Duration{Duration:50000000}}, Controller:server.ControllerConfig{DegradedModeWaitDuration:typeutil.Duration{Duration:0}, LTBMaxWaitDuration:typeutil.Duration{Duration:30000000000}, RequestUnit:server.RequestUnitConfig{ReadBaseCost:0.125, ReadPerBatchBaseCost:0.5, ReadCostPerByte:1.52587[890](https://github.com/tikv/pd/actions/runs/7536646974/job/20514367350#step:5:891)625e-05, WriteBaseCost:1, WritePerBatchBaseCost:1, WriteCostPerByte:0.0009765625, CPUMsCost:0.3333333333333333}, EnableControllerTraceLog:false}}
            	            	actual  : &config.Config{ClientUrls:"http://127.0.0.1:35273/", PeerUrls:"http://127.0.0.1:39959/", AdvertiseClientUrls:"http://127.0.0.1:35273/", AdvertisePeerUrls:"http://127.0.0.1:39959/", Name:"pd1", DataDir:"/tmp/pd-tests1161716090", ForceNewCluster:false, EnableGRPCGateway:true, InitialCluster:"pd1=http://127.0.0.1:39959/", InitialClusterState:"new", InitialClusterToken:"pd-cluster", Join:"", LeaderLease:3, Log:log.Config{Level:"warn", Format:"text", DisableTimestamp:false, File:log.FileLogConfig{Filename:"", MaxSize:0, MaxDays:0, MaxBackups:0}, Development:false, DisableCaller:false, DisableStacktrace:false, DisableErrorVerbose:true, Sampling:(*zap.SamplingConfig)(nil), ErrorOutputPath:""}, LogFileDeprecated:"", LogLevelDeprecated:"", MaxConcurrentTSOProxyStreamings:5000, TSOProxyRecvFromClientTimeout:typeutil.Duration{Duration:3600000000000}, TSOSaveInterval:typeutil.Duration{Duration:3000000000}, TSOUpdatePhysicalInterval:typeutil.Duration{Duration:50000000}, EnableLocalTSO:false, Metric:metricutil.MetricConfig{PushJob:"pd1", PushAddress:"", PushInterval:typeutil.Duration{Duration:15000000000}}, Schedule:config.ScheduleConfig{MaxSnapshotCount:0x40, MaxPendingPeerCount:0x40, MaxMergeRegionSize:0x14, MaxMergeRegionKeys:0x30d40, SplitMergeInterval:typeutil.Duration{Duration:3600000000000}, SwitchWitnessInterval:typeutil.Duration{Duration:3600000000000}, EnableOneWayMerge:false, EnableCrossTableMerge:true, PatrolRegionInterval:typeutil.Duration{Duration:10000000}, MaxStoreDownTime:typeutil.Duration{Duration:1800000000000}, MaxStorePreparingTime:typeutil.Duration{Duration:172800000000000}, LeaderScheduleLimit:0x4, LeaderSchedulePolicy:"count", RegionScheduleLimit:0xa, WitnessScheduleLimit:0x4, ReplicaScheduleLimit:0x40, MergeScheduleLimit:0x8, HotRegionScheduleLimit:0x4, HotRegionCacheHitsThreshold:0x3, StoreBalanceRate:0, StoreLimit:map[uint64]config.StoreLimitConfig{}, TolerantSizeRatio:2.5, LowSpaceRatio:0.8, HighSpaceRatio:0.7, RegionScoreFormulaVersion:"v2", SchedulerMaxWaitingOperator:0x5, DisableLearner:false, DisableRemoveDownReplica:false, DisableReplaceOfflineReplica:false, DisableMakeUpReplica:false, DisableRemoveExtraReplica:false, DisableLocationReplacement:false, EnableRemoveDownReplica:true, EnableReplaceOfflineReplica:true, EnableMakeUpReplica:true, EnableRemoveExtraReplica:true, EnableLocationReplacement:true, EnableDebugMetrics:false, EnableJointConsensus:true, EnableTiKVSplitRegion:true, Schedulers:config.SchedulerConfigs{config.SchedulerConfig{Type:"balance-region", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"balance-leader", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"balance-witness", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"hot-region", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"transfer-witness-leader", Args:[]string(nil), Disable:false, ArgsPayload:""}, config.SchedulerConfig{Type:"evict-slow-store", Args:[]string(nil), Disable:false, ArgsPayload:""}}, SchedulersPayload:map[string]interface {}{"balance-hot-region-scheduler":map[string]interface {}{"byte-rate-rank-step-ratio":0.05, "count-rank-step-ratio":0.01, "dst-tolerance-ratio":1.05, "enable-for-tiflash":"true", "forbid-rw-type":"none", "great-dec-ratio":0.95, "key-rate-rank-step-ratio":0.05, "max-peer-number":1000, "max-zombie-rounds":3, "min-hot-byte-rate":100, "min-hot-key-rate":10, "min-hot-query-rate":10, "minor-dec-ratio":0.99, "query-rate-rank-step-ratio":0.05, "rank-formula-version":"v2", "read-priorities":[]interface {}{"query", "byte"}, "split-thresholds":0.2, "src-tolerance-ratio":1.05, "strict-picking-store":"true", "write-leader-priorities":[]interface {}{"query", "byte"}, "write-peer-priorities":[]interface {}{"byte", "key"}}, "balance-leader-scheduler":map[string]interface {}{"batch":4, "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "balance-region-scheduler":map[string]interface {}{"name":"balance-region-scheduler", "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "balance-witness-scheduler":map[string]interface {}{"batch":4, "ranges":[]interface {}{map[string]interface {}{"end-key":"", "start-key":""}}}, "evict-slow-store-scheduler":map[string]interface {}{"evict-stores":[]interface {}{}, "recovery-duration":600}, "transfer-witness-leader-scheduler":interface {}(nil)}, HotRegionsWriteInterval:typeutil.Duration{Duration:600000000000}, HotRegionsReservedDays:0x7, MaxMovableHotPeerSize:512, EnableDiagnostic:true, EnableWitness:false, SlowStoreEvictingAffectedStoreRatioThreshold:0.3, StoreLimitVersion:"v1", HaltScheduling:false}, Replication:config.ReplicationConfig{MaxReplicas:0x5, LocationLabels:typeutil.StringSlice{"zone", "rack"}, StrictlyMatchLabel:false, EnablePlacementRules:true, EnablePlacementRulesCache:false, IsolationLevel:""}, PDServerCfg:config.PDServerConfig{UseRegionStorage:true, MaxResetTSGap:typeutil.Duration{Duration:86400000000000}, KeyType:"table", RuntimeServices:typeutil.StringSlice{}, MetricStorage:"http://127.0.0.1:1234/", DashboardAddress:"http://127.0.0.1:35273/", TraceRegionFlow:true, FlowRoundByDigit:3, MinResolvedTSPersistenceInterval:typeutil.Duration{Duration:1000000000}, ServerMemoryLimit:0, ServerMemoryLimitGCTrigger:0.7, EnableGOGCTuner:false, GCTunerThreshold:0.6, BlockSafePointV1:false}, ClusterVersion:semver.Version{Major:4, Minor:0, Patch:0, PreRelease:"beta", Metadata:""}, Labels:map[string]string{}, QuotaBackendBytes:0x200000000, AutoCompactionMode:"periodic", AutoCompactionRetention:"1h", TickInterval:typeutil.Duration{Duration:500000000}, ElectionInterval:typeutil.Duration{Duration:3000000000}, PreVote:true, MaxRequestBytes:0x9600000, Security:configutil.SecurityConfig{TLSConfig:grpcutil.TLSConfig{CAPath:"", CertPath:"", KeyPath:"", CertAllowedCN:[]string(nil), SSLCABytes:[]uint8(nil), SSLCertBytes:[]uint8(nil), SSLKEYBytes:[]uint8(nil)}, RedactInfoLog:false, Encryption:encryption.Config{DataEncryptionMethod:"plaintext", DataKeyRotationPeriod:typeutil.Duration{Duration:604800000000000}, MasterKey:encryption.MasterKeyConfig{Type:"plaintext", MasterKeyKMSConfig:encryption.MasterKeyKMSConfig{KmsKeyID:"", KmsRegion:"", KmsEndpoint:""}, MasterKeyFileConfig:encryption.MasterKeyFileConfig{FilePath:""}}}}, LabelProperty:config.LabelPropertyConfig{}, WarningMsgs:[]string(nil), DisableStrictReconfigCheck:false, HeartbeatStreamBindInterval:typeutil.Duration{Duration:60000000000}, LeaderPriorityCheckInterval:typeutil.Duration{Duration:60000000000}, Logger:(*zap.Logger)(nil), LogProps:(*log.ZapProperties)(nil), Dashboard:config.DashboardConfig{TiDBCAPath:"", TiDBCertPath:"", TiDBKeyPath:"", PublicPathPrefix:"", InternalProxy:false, EnableTelemetry:false, EnableExperimental:false}, ReplicationMode:config.ReplicationModeConfig{ReplicationMode:"dr-auto-sync", DRAutoSync:config.DRAutoSyncReplicationConfig{LabelKey:"foobar", Primary:"", DR:"", PrimaryReplicas:0, DRReplicas:0, WaitStoreTimeout:typeutil.Duration{Duration:60000000000}, WaitRecoverTimeout:typeutil.Duration{Duration:0}, PauseRegionSplit:false}}, Keyspace:config.KeyspaceConfig{PreAlloc:[]string(nil), WaitRegionSplit:true, WaitRegionSplitTimeout:typeutil.Duration{Duration:30000000000}, CheckRegionSplitInterval:typeutil.Duration{Duration:50000000}}, Controller:server.ControllerConfig{DegradedModeWaitDuration:typeutil.Duration{Duration:0}, LTBMaxWaitDuration:typeutil.Duration{Duration:30000000000}, RequestUnit:server.RequestUnitConfig{ReadBaseCost:0.125, ReadPerBatchBaseCost:0.5, ReadCostPerByte:1.525878[906](https://github.com/tikv/pd/actions/runs/7536646974/job/20514367350#step:5:907)25e-05, WriteBaseCost:1, WritePerBatchBaseCost:1, WriteCostPerByte:0.0009765625, CPUMsCost:0.3333333333333333}, EnableControllerTraceLog:false}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -102,3 +102,3 @@
            	            	   EnableJointConsensus: (bool) true,
            	            	-  EnableTiKVSplitRegion: (bool) false,
            	            	+  EnableTiKVSplitRegion: (bool) true,
            	            	   Schedulers: (config.SchedulerConfigs) (len=6) {
            	            	@@ -222,4 +222,4 @@
            	            	   LocationLabels: (typeutil.StringSlice) (len=2) {
            	            	-   (string) (len=3) "idc",
            	            	-   (string) (len=4) "host"
            	            	+   (string) (len=4) "zone",
            	            	+   (string) (len=4) "rack"
            	            	   },
            	Test:       	TestConfigTestSuite/TestConfigAll
```
### CI link
https://github.com/tikv/pd/actions/runs/7536646974/job/20514367350

### Reason for failure (if possible)


### Anything else

`

--- a/tests/server/config/config_test.go
+++ b/tests/server/config/config_test.go
@@ -101,6 +101,7 @@ func (suite *configTestSuite) SetupSuite() {
 func (suite *configTestSuite) TearDownSuite() {
 	suite.env.Cleanup()
 }
+
 func (suite *configTestSuite) TestConfigAll() {
 	suite.env.RunTestInTwoModes(suite.checkConfigAll)
 }
@@ -140,16 +141,17 @@ func (suite *configTestSuite) checkConfigAll(cluster *tests.TestCluster) {
 	re.NoError(err)
 	err = tu.CheckPostJSON(testDialClient, addr, postData, tu.StatusOK(re))
 	re.NoError(err)
-
-	newCfg := &config.Config{}
-	err = tu.ReadGetJSON(re, testDialClient, addr, newCfg)
-	re.NoError(err)
 	cfg.Replication.MaxReplicas = 5
 	cfg.Replication.LocationLabels = []string{"zone", "rack"}
 	cfg.Schedule.RegionScheduleLimit = 10
 	cfg.PDServerCfg.MetricStorage = "http://127.0.0.1:9090"
-	re.Equal(newCfg, cfg)
 
+	tu.Eventually(re, func() bool {
+		newCfg := &config.Config{}
+		err = tu.ReadGetJSON(re, testDialClient, addr, newCfg)
+		re.NoError(err)
+		return suite.Equal(newCfg, cfg)
+	})
 	// the new way
 	l = map[string]any{
 		"schedule.tolerant-size-ratio":            2.5,
@@ -165,9 +167,6 @@ func (suite *configTestSuite) checkConfigAll(cluster *tests.TestCluster) {
 	re.NoError(err)
 	err = tu.CheckPostJSON(testDialClient, addr, postData, tu.StatusOK(re))
 	re.NoError(err)
-	newCfg1 := &config.Config{}
-	err = tu.ReadGetJSON(re, testDialClient, addr, newCfg1)
-	re.NoError(err)
 	cfg.Schedule.EnableTiKVSplitRegion = false
 	cfg.Schedule.TolerantSizeRatio = 2.5
 	cfg.Replication.LocationLabels = []string{"idc", "host"}
@@ -178,7 +177,12 @@ func (suite *configTestSuite) checkConfigAll(cluster *tests.TestCluster) {
 	v, err := versioninfo.ParseVersion("v4.0.0-beta")
 	re.NoError(err)
 	cfg.ClusterVersion = *v
-	re.Equal(cfg, newCfg1)
+	tu.Eventually(re, func() bool {
+		newCfg1 := &config.Config{}
+		err = tu.ReadGetJSON(re, testDialClient, addr, newCfg1)
+		re.NoError(err)
+		return suite.Equal(cfg, newCfg1)
+	})
 
 	// revert this to avoid it affects TestConfigTTL
 	l["schedule.enable-tikv-split-region"] = "true"

